defaults/main.yml

```diff
@@ -1,260 +1,355 @@
-ifupdown_reconfigure_auto: True
-
-
-# .. envvar:: ifupdown_reconfigure_auto_file
-#
-# Name of the file which, if present, tells the reconfiguration script to not
-# make changes in live network interfaces.
-ifupdown_reconfigure_auto_file: '/run/network/ifupdown-manual'
-
-
-# .. envvar:: ifupdown_reconfigure_script
-#
-# Path where reconfiguration script is stored.
-ifupdown_reconfigure_script: '{{ (ansible_local.root.lib
-                                 if (ansible_local|d() and ansible_local.root|d()
-                                     and ansible_local.root.lib|d())
-                                 else "/usr/local/lib") +
-                                 "/ifupdown-reconfigure-interfaces" }}'
-
-
-# .. envvar:: ifupdown_reconfigure_init_file
-#
-# Name of the file that contains an initial state of the ifupdown configuration
-# and indicates that all network interfaces should be stopped before
-# configuration.
-ifupdown_reconfigure_init_file: '/etc/network/interfaces.d/old-interfaces'
-
-
-# .. envvar:: ifupdown_apply_config
-#
-# Apply generated interface configuration immediately. If set to False, the role
-# will generate a script in the :file:`/tmp/` directory which will apply the configuration
-# at the end of the current Ansible play.
-ifupdown_apply_config: True
-
-
-# .. envvar:: ifupdown_apply_config_pre_script
-#
-# Shell commands in YAML text block format, which will be executed by the
-# config script before interface reconfiguration.
-ifupdown_apply_config_pre_script: ''
-
-
-# .. envvar:: ifupdown_apply_config_post_script
 #
-# Shell commands in YAML text block format, which will be executed by the
-# config script after interface reconfiguration.
-ifupdown_apply_config_post_script: ''
+# .. include:: includes/all.rst
 
 
-# --------------------------------
-#   Network interface definition
-# --------------------------------
+# APT packages [[[
+# ----------------
 
-# .. envvar:: ifupdown_default_config
+# .. envvar:: ifupdown__base_packages [[[
 #
-# Which default configuration from var/ directory should be used? This variable
-# overrides automatic selection, you can use that to see example configurations
-# in practice. Specify a filename without '.yml' extension.
-ifupdown_default_config: ""
-
+# List of base APT packages to install.
+ifupdown__base_packages: [ 'resolvconf', 'rdnssd', 'bsdutils', 'rsync' ]
 
-# .. envvar:: ifupdown_external_interface
+                                                                   # ]]]
+# .. envvar:: ifupdown__dynamic_packages [[[
 #
-# Default external network interface.
-ifupdown_external_interface: 'eth0'
-
+# List of APT packages to install based on the requirements of the interface
+# types that are configured by the role (support for bridges, bonding, VLANs,
+# depends on the presence of a given interface type).
+ifupdown__dynamic_packages: '{{ lookup("template", "lookup/ifupdown__dynamic_packages.j2") | string | from_yaml }}'
```

Comment:
`convert_data=False` works :wink: So Ansible folks are aware of this. Fixed in my working copy.